compiler/optimizer/OSRDefAnalysis.cpp

```diff
@@ -1314,6 +1314,35 @@ TR_OSRLiveRangeAnalysis::optDetailString() const throw()
    return "O^O OSR LIVE RANGE ANALYSIS: ";
    }
 
+bool TR_OSRExceptionEdgeRemoval::addDeadStores(TR::Block* osrBlock, TR_BitVector& dead, bool inited)
+   {
+   _seenDeadStores->empty();
+   for (TR::TreeTop *tt = osrBlock->getFirstRealTreeTop(); tt != osrBlock->getExit(); tt = tt->getNextTreeTop())
+      {
+      TR::Node *node = tt->getNode();
+      if (node->getOpCode().isStoreDirect() && node->storedValueIsIrrelevant())
+         _seenDeadStores->set(node->getSymbolReference()->getReferenceNumber());
+      }
+
+   if (inited)
+      dead &= *_seenDeadStores;
+   else
+      dead |= *_seenDeadStores;
+
+   return !_seenDeadStores->isEmpty();
+   }
+
+void TR_OSRExceptionEdgeRemoval::removeDeadStores(TR::Block* osrBlock, TR_BitVector& dead)
+   {
```

Comment:
An assert to enforce the correct kind of bloc is given would be a good idea here...

Reply:
Added an assert for both helper functions.